### DISCUSSION TITLE
Added: LDM notes for April 3rd, 2019

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#2414] [General]
https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-04-03.md

**Comment by YairHalberstadt** [comment]
@agocke @gafter 

> 3. No new syntax, do matching in two passes. First, look if the old form (where T : struct) is present and use it if present. Second, expand to look for both forms if no match was found.
>
> ...
> **Conclusion**
> We like option 3. This would maintain backwards compatibility and is a simple explanation for what T? means in OHI.

This seems at odds with https://github.com/dotnet/csharplang/issues/2378#issuecomment-479634969

> We discussed this 2019-04-03 at the LDM. What we came away with are the following resolutions:
> 1. In an override or implementation (that lacks constraints), T? would always mean Nullable<T>. This "fix" will be implemented ASAP to fix the product compatibility bug. The other steps below would be done separately.
> 2. If you want to use a nullable reference type annotation on a type parameter (e.g. T?) in an override or implementation, you would be required to write the constraint where T: class in the override or implementation. The ability to write that constraint is a new feature.
> 3. We would also allow where T: struct in an override or implementation for symmetry.

Could you please clarify this?
Thanks

**Comment by agocke** [comment]
Thanks that should read option **2**, that was a typo. Fixed.
